invenio_rdm_records/records/models.py

```diff
@@ -54,6 +56,13 @@ class RDMRecordMetadata(db.Model, RecordMetadataBase, ParentRecordMixin):
     media_bucket_id = db.Column(UUIDType, db.ForeignKey(Bucket.id))
     media_bucket = db.relationship(Bucket, foreign_keys=[media_bucket_id])
 
+    # The deletion status is stored in the model so that we can use it in SQL queries
+    deletion_status = db.Column(
+        ChoiceType(RecordDeletionStatusEnum, impl=db.String(1)),
+        nullable=False,
+        default=RecordDeletionStatusEnum.PUBLISHED.value,
```

Comment:
```suggestion
        default=RecordDeletionStatusEnum.PUBLISHED,
```

question/nit: is it still needed now that we're using `ChoiceType`? I know it works probably, since the field is a bit more flexible with the accepted values, but just to confirm and advertise it a bit in case one wonders how to set the field.

Reply:
I've tried out storing the enum itself rather than the `.value`, and it does work nicely.
However, this will also cause the enum value rather than the string to be returned when accessing this model field, which is not expected in a number of places. :boom: 

Also, the alembic migration falls back to the string `"P"` if all other heuristics fail to fetch a default value.
So in order to stay consistent, I reverted back to setting the `PUBLISHED.value` rather than `PUBLISHED` here.

Reply:
Ouch, yeah that will definitely cause issues... I though by using the Enum, we're making our lives easier when doing comparisons/conversions across the codebase, i.e. `"P" == RecordDeletionStatusEnum.PUBLISHED`. 

What are the places that expect a string value coming back from the model? Is it serialization/indexing? It feels a bit dirty to rely on the "temporary" value that the model field is set to, i.e. if the following behaviour doesn't fail we're in trouble:

```python
record.deletion_status = "P"
# using "is" as in the exact value + type without implicit equality checks
assert record.deletion_status is "P"
assert record.deletion_status is not RecordDeletionStatusEnum.PUBLISHED

record.deletion_status = RecordDeletionStatusEnum.PUBLISHED
assert record.deletion_status is RecordDeletionStatusEnum.PUBLISHED
assert record.deletion_status is not "P"
```

I would expect something like:

```python
record.deletion_status = "P"
assert record.deletion_status == "P"
assert record.deletion_status == RecordDeletionStatusEnum.PUBLISHED
assert record.deletion_status is not "P"
assert record.deletion_status is RecordDeletionStatusEnum.PUBLISHED

record.deletion_status = RecordDeletionStatusEnum.PUBLISHED
assert record.deletion_status == RecordDeletionStatusEnum.PUBLISHED
assert record.deletion_status is not "P"
assert record.deletion_status is RecordDeletionStatusEnum.PUBLISHED
```

Reply:
No, unfortunately it doesn't seem that comparisons between enums and strings are so easily possible:
```python
In [9]: RecordDeletionStatusEnum.DELETED.value
Out[9]: 'D'

In [10]: RecordDeletionStatusEnum.DELETED == "D"
Out[10]: False
```

However, checks similar to the following are possible (note that this will raise an exception if `"D"` isn't a valid enum value):
```python
In [14]: RecordDeletionStatusEnum("D") is RecordDeletionStatusEnum.DELETED
Out[14]: True
```

---

The enums are mostly used as a backbone in the implementation, to make sure no invalid values can be set in the model; they're used as the basis for normalization while _setting_ the record deletion status.
For _getting_ the deletion status, I usually provide several options:
* `record.deletion_status` gives back a `RecordDeletionStatus` object (calling `str()` on that will return the enum string value)
* `record.deletion_status.status` gives back the string value
* `record.deletion_status._status` gives back the `RecordDeletionStatusEnum` object

:heavy_plus_sign: I've just overridden the equality checks for the **system field**, s.t. it can handle comparisons with strings and enum values as expected.

---

:information_source:  **Update**
I've just experimented a bit, and it seems like the `ChoiceField` (in the **model**) accepts both strings as well as enum values.
When setting the field to a string (e.g. `"P"`), it will have the string value for the remainder of the session.
In a new session, the value will be normalized into an enum value:

```python
In [24]: r.model.deletion_status = "P"

In [25]: r.model.deletion_status
Out[25]: 'P'

In [26]: db.session.commit()

In [27]: r = rec_svc.record_cls.get_record(r.id)

In [28]: r.model.deletion_status
Out[28]: <RecordDeletionStatusEnum.PUBLISHED: 'P'>
```

Note that `db.session.commit()` will raise an exception (`StatementError`) if the value is set to something invalid.

Although it should be pretty rare for anybody to interact with the model's DB column directly, usually it should go through the system field.
And as mentioned above, that takes care of conversions and normalization s.t. the developers don't have to.

Reply:
Ok, that makes sense. Thanks a lot for digging into this, I learned something new as well today :)

In that case, since we're always going through the system field for everything (setting, serialization, etc.), we should be good. I don't think there's a need to explicitly check for all the weird scenarios I mentioned in https://github.com/inveniosoftware/invenio-rdm-records/pull/1386#discussion_r1289998046 (since there's no point in verifying the `ChoiceType`'s behavior).